### ISSUE TITLE
string-distance panic on deletion

### ISSUE THREAD

**Issue by CeleritasCelery** [CeleritasCelery/rune#75]
@Ki11erRabbit I was adding some [more tests](https://docs.rs/distance/latest/src/distance/levenshtein.rs.html#70) for `string-distance` and I found a case the causes a panic.

```rust
 assert_eq!(3, levenshtein_distance("kitten".chars(), "sitting".chars()));
```

**Comment by Ki11erRabbit** [comment]
I'll look into it.

**Comment by Ki11erRabbit** [comment]
I have fixed the issue in #76 as well as added some documentation to the internals of the `levenshtein` function.
